packages/angular_devkit/build_angular/src/app-shell/index.ts

```diff
@@ -161,9 +169,21 @@ export class AppShellBuilder implements Builder<BuildWebpackAppShellSchema> {
           })
           .then((html: string) => {
             return this.context.host
-              .write(outputPath, virtualFs.stringToFileBuffer(html))
+              .write(outputIndexPath, virtualFs.stringToFileBuffer(html))
               .toPromise();
           })
+          .then(() => {
+            if (browserOptions.serviceWorker) {
+              return augmentAppWithServiceWorker(
```

Comment:
I wonder if re-running this would introduce a noticeable delay for large projects (with lots of files to hash).

Reply:
That's a concern, yeah. I don't think `ngsw` has a way to just update the hashes for some files though. I don't want to duplicate the hashing logic it uses inside the CLI.

Reply:
I'm not sure this is worth optimizing just yet.

Honestly, the "correct" solution to me is to not generate `ngsw.json` until the entire build is done, including the App Shell. Running it twice seems like a hack (but a hack that will work).